File: src/main/java/org/bundleproject/discover/utils/StringUtils.java
package org.bundleproject.discover.utils;

public class StringUtils {

    public static String wrapText(String text, int charLength) {
        StringBuilder sb = new StringBuilder();
        int lineLength = 0;
        boolean needsLineBreak = false;
        for (char c : text.toCharArray()) {
            lineLength += 1;
            if (c == '\n') lineLength = 0;
            if (lineLength > charLength) {
                needsLineBreak = true;
            }
            if (needsLineBreak && c == ' ') {
                lineLength = 0;
                sb.append('\n');
                needsLineBreak = false;
            } else {
                sb.append(c);
            }
        }

        return sb.toString();
    }

}

File: src/main/kotlin/org/bundleproject/discover/utils/StringUtils.kt
package org.bundleproject.discover.utils

object StringUtils {
    fun wrapText(text: String, charLength: Int): String {
        val sb = StringBuilder()
        var lineLength = 0
        var needsLineBreak = false
        for (c in text.toCharArray()) {
            lineLength += 1
            if (c == '\n') lineLength = 0
            if (lineLength > charLength) {
                needsLineBreak = true
            }
            if (needsLineBreak && c == ' ') {
                lineLength = 0
                sb.append('\n')
                needsLineBreak = false
            } else {
                sb.append(c)
            }
        }
        return sb.toString()
    }
}
